### DISCUSSION TITLE
Application Installation Failed

### DISCUSSION THREAD

**Discussion by baqwas** [FIRST-Tech-Challenge/FtcRobotController#503] [General] [answered]
Looking for _newbie_ guidance, please!

My fork didn't sync properly with SDK 8.1 (as remote) despite prior successes (7.2 & 8.0). The classic error message is:

_The device already has a newer version of the application..._

My fork master is not behind the upstream FIRST-Tech-Challenge master. Of course, there are a few branches under my fork master.

**build.dependencies.gradle** has the requisite 8.1.0 entries as shown below:
```
repositories {
    mavenCentral()
    google() // Needed for androidx
    flatDir {
        dirs rootProject.file('libs')
    }
    // 2022-10-07 armw 1 - FTC Dashboard & Road Runner basic installation
    maven { url = 'https://maven.brott.dev/' }
}

dependencies {
    implementation 'org.firstinspires.ftc:Inspection:8.1.0'
    implementation 'org.firstinspires.ftc:Blocks:8.1.0'
    implementation 'org.firstinspires.ftc:Tfod:8.1.0'
    implementation 'org.firstinspires.ftc:RobotCore:8.1.0'
    implementation 'org.firstinspires.ftc:RobotServer:8.1.0'
    implementation 'org.firstinspires.ftc:OnBotJava:8.1.0'
    implementation 'org.firstinspires.ftc:Hardware:8.1.0'
    implementation 'org.firstinspires.ftc:FtcCommon:8.1.0'
    implementation 'org.tensorflow:tensorflow-lite-task-vision:0.2.0'
    implementation 'androidx.appcompat:appcompat:1.5.1'
    implementation 'org.firstinspires.ftc:gameAssets-PowerPlay:1.0.0'

    // 2022-10-07 armw 2 - basic installation
    implementation 'com.acmerobotics.dashboard:dashboard:0.4.6'
}

```
Also, among other declarations, **build.common.gradle** has:

```
minSdkVersion 24
targetSdkVersion 28
    ndkVersion '21.3.6528147'

```

**build.gradle(:TeamCode)** has:
```
apply from: '../build.common.gradle'
apply from: '../build.dependencies.gradle'

android {
    namespace = 'org.firstinspires.ftc.teamcode'
}

dependencies {
    implementation project(':FtcRobotController')
    implementation 'com.google.android.datatransport:transport-runtime:3.1.8'
    annotationProcessor files('lib/OpModeAnnotationProcessor.jar')
    implementation 'org.ftclib.ftclib:core:2.0.1' // core https://docs.ftclib.org/ftclib/installation
    //implementation 'com.google.code.gson:gson:2.8.5' // armw 2022-09-22

    // 2022-10-07 armw Road Runner Core (FTC)
    // https://rr.brott.dev/docs/v1-0/installation/
    implementation 'com.acmerobotics.roadrunner:core:1.0.0-beta0'

    // 2022-11-10 armw OpenFTC/EasyOpenCV
    // https://github.com/OpenFTC/EasyOpenCV
    implementation 'org.openftc:easyopencv:1.5.2'

    // 2022-11-10 armw EOCV-AprilTag-Plugin
    // https://github.com/OpenFTC/EOCV-AprilTag-Plugin
    implementation 'org.openftc:apriltag:1.1.0'
}
```

What is the recommendation for a _newbie_ to recover from this installation error assuming that all TeamCode OpModes can be re-written from scratch? I've checked the remote fork and ensured that all files did sync. At my end, none of the branches will deploy even after the following operations:

```
git branch feature1
git pull origin master
```
followed by build and deploy attempts - Build works without errors, deploy fails with error message about newer version already on device.

I simply want to ensure that I can deploy OpModes using SDK 8.1.

Of course, my understanding of GitHub operations is very limited to say the least. All suggestions on being able to stay in sync with the remote fork (especially with 8.1.x or 8.2 forthcoming) would be sincerely appreciated. Thanks.

Kind regards.

**Comment by cmacfarl** [comment, marked as answer]
Did you make any changes to AndroidManifest.xml?  

https://github.com/FIRST-Tech-Challenge/FtcRobotController/blob/master/FtcRobotController/src/main/AndroidManifest.xml

**Reply by baqwas** [reply]
Many, many thanks, @cmacfarl!

My version has:
```
    android:versionCode="47"
          android:versionName="8.0">
```
Not skilled (or _brave_) enough to make these changes at will but I've been at the other end of support too where a customer or two often declare "I didn't change a thing."

I believe manually reverting the changes to your example should set me straight? Thanks.

**Reply by cmacfarl** [reply]
So I looked at your fork and it has the proper versionCode.  Specifically this line.

https://github.com/baqwas/FtcRobotController/blob/master/FtcRobotController/src/main/AndroidManifest.xml#L4

What you posted above is 47, instead of the 48, and yes, if you try to deploy that, and there's already an instance of a RC with a versionCode of 48, AndroidStudio or adb will complain.  If that is a local change on your laptop, and it has not been committed, I'd just do a git restore on that file and everything should be fine.  But it does beg the question as to why your manifest file is a version behind.

**Reply by baqwas** [reply]
Everything is fine with the fork now! We've been in overdrive mode. We'll benefit from the reorientation of the IMU. Thanks again.

Kind regards.
